nbconvert/exporters/templateexporter.py

```diff
@@ -283,6 +284,8 @@ def _create_environment(self):
             extensions=JINJA_EXTENSIONS
             )
 
+        environment.globals['uuid4'] = uuid.uuid4
```

Comment:
I tried putting this in filters, but the `set` statement in the template didn't appear to look in that namespace. I'm happy to make a more general apparatus in a separate PR for adding things to the `globals` namespace if we think that's useful.